girder/molecules/server/experiment.py

```diff
@@ -80,15 +81,29 @@ def create(self, body):
     def find_experiment(self, params):
         user = getCurrentUser()
 
-        query = { }
-        if 'molecularFormula' in params:
-            query['molecularFormula'] = params['molecularFormula']
+        if 'source' in params.keys():
+            from .nist import search_nist_inchi, get_jdx
+            from jcamp import jcamp_read
+
+            inchi = params['molecularFormula']      # TODO change API to accept
+                                                    # a different key. From
+                                                    # molecularFormula to inchi?
```

Comment:
Yes, lets allow inchi to be passed in it own parameter.

Reply:
I was checking that yesterday, but have not found the way of doing it. I will keep lurking into the code to understand better.

Reply:
You should just be able to add it as a query parameter when you make your request [here](https://github.com/muammar/openchemistrypy/blob/ir_spec/openchemistry/__init__.py#L790)

Reply:
I tried doing that but the API threw an error that the key did not exist. I will try to reproduce it to post the traceback here.

Reply:
You may have to add a check to see which key exists.

Reply:
Feel free to push a WIP commit and I can see what is going on.

Reply:
Oh, I see now. I misspelled something. Now it works.  This is done, too now.